### PR TITLE
Update homogenization engine

### PR DESCRIPTION
**Pull request by vlukes**
There are the following important points:

- new storing of time variable correctors
- remove obsolete `verify_correctors()` in `TCorrectorsViaPressureEVP`
- remove `dump_*` options from the engine. It is replaced by  `save_formats` option, e.g.:
`'save_formats': ['vtk', 'h5'],`
- reimplementation of matching periodic planes. Now, it allows for skew geometries:
`match_plane_by_dir(coors1, coors2, direction)`
where `direction` is a direction in which `coors2` are shifted regarding to `coors1`

### CODE REVIEW

File: sfepy/homogenization/coefs_base.py [hunk 1]
```diff
@@ -113,83 +113,92 @@ def iter_solutions(self):
         else:
             yield '', self.state
 
+    def iter_time_steps(self):
+        if hasattr(self, 'n_tstep') and self.n_tstep > 0:
```
Comment (rc) [inline sfepy/homogenization/coefs_base.py]: Why not n_step as in time-stepping solvers?

File: sfepy/homogenization/coefs_base.py [hunk 2]
```diff
@@ -113,83 +113,92 @@ def iter_solutions(self):
...
+                yield self.get_ts_val(ii)
+        else:
+            yield self
+
+    def get_ts_val(self, step):
+        if hasattr(self, 'states'):
+            states = nm.zeros(self.states.shape, dtype=nm.object)
+            for idx in self.components:
+                state = {k: v[step] for k, v in\
+                            six.iteritems(self.states[idx])}
```
Comment (rc) [inline sfepy/homogenization/coefs_base.py]: Indentation?

File: sfepy/homogenization/coefs_base.py [hunk 3]
```diff
@@ -113,83 +113,92 @@ def iter_solutions(self):
...
-                                 var.n_components)
-                        out[skey] = Struct(name = 'dump', mode = 'vertex',
-                                           data = dof_vector,
-                                           dofs = var.dofs,
-                                           shape = shape,
-                                           var_name = var_name)
+                    skey = var_name + '_' + key if key else var_name
+                    var = variables[vname]
+                    shape = (var.n_dof // var.n_components,
+                                var.n_components)
```
Comment (rc) [inline sfepy/homogenization/coefs_base.py]: Indentation?

File: sfepy/homogenization/coefs_base.py [hunk 4]
```diff
@@ -113,83 +113,92 @@ def iter_solutions(self):
...
+                    out[skey] = Struct(name = 'dump', mode = 'vertex',
+                                       data = dof_vector,
+                                       shape = shape,
+                                       var_name = vname)
```
Comment (rc) [inline sfepy/homogenization/coefs_base.py]: ' = ' -> '='

**Comment by rc** [comment]
Thanks! Besides the small issues in the comments attached to the code, let us not forget, that no sfepy format saves higher order DOFs yet. That was the original reason for the "dumps". Overall I agree with this PR, but we should not forget that an address it by improving out hdf5-based output.

**Comment by rc** [comment]
I am not sure why the travis tests failed  - might it be again related to multiprocessing?

**Comment by vlukes** [comment]
The small issues, e.g. indentation, `n_tsteps` -> 'n_steps', ...,  are solved.

I have no idea  why the travis tests failed.

**Comment by rc** [comment]
Thanks, travis now passes, so it might have been a temporary glitch.

I have noticed other lines with bad indentation related to sd_names -> save_names change, but if you do not get to it today, I will merge this evening anyway - it is not that important.

**Comment by rc** [comment]
Thanks for the update! All tests pass, but

```
python ./script/tile_periodic_mesh.py -r 2,2 meshes/elements/2_4_2.mesh out-per.mesh
```

fails. Would you prefere fixing that here, or a new PR, after this is merged?

**Comment by vlukes** [comment]
I'll have a look at it.

**Comment by vlukes** [comment]
Now, it works.

**Comment by rc** [comment]
Yes, thanks!

**Pull request merged**